python_modules/libraries/dagster-dbt/dagster_dbt/dagster_dbt_translator.py

```diff
@@ -115,6 +116,35 @@ def get_asset_key(cls, dbt_resource_props: Mapping[str, Any]) -> AssetKey:
         """
         return default_asset_key_fn(dbt_resource_props)
 
+    @classmethod
+    @public
+    def get_partition_mapping(
+        cls,
+        dbt_resource_props: Mapping[str, Any],
+        dbt_parent_resource_props: Mapping[str, Any],
+    ) -> Optional[PartitionMapping]:
+        """A function that takes two dictionaries: the first, representing properties of a dbt
+        resource; and the second, representing the properties of a parent dependency to the first
+        dbt resource. The function returns the Dagster partition mapping for the dbt dependency.
+
+        Note that a dbt resource is unrelated to Dagster's resource concept, and simply represents
+        a model, seed, snapshot or source in a given dbt project. You can learn more about dbt
+        resources and the properties available in this dictionary here:
+        https://docs.getdbt.com/reference/artifacts/manifest-json#resource-details
+
+        This method can be overridden to provide a custom partition mapping for a dbt dependency.
+
+        Args:
+            dbt_resource_props (Mapping[str, Any]):
+                A dictionary representing the dbt child resource.
+            dbt_parent_resource_props (Mapping[str, Any]):
+                A dictionary representing the dbt parent resource, in relationship to the child.
+
+        Returns:
+            Optional[PartitionMapping]: The Dagster partition mapping for the dbt resource.
```

Comment:
I think it's worth clarifying what returning ` None` means. I assume it means we'll use the default partition mapping?